### DISCUSSION TITLE
One time self destructing links in adonisJS 5

### DISCUSSION THREAD

**Discussion by zmaglica** [adonisjs/core#1367] [Help] [answered]
Hi,

is there a way to create one time (self-destructing) links in adonis 5? This can be useful on password resets or invitation links

**Comment by thetutlage** [comment, marked as answer]
We have [signed urls](https://preview.adonisjs.com/guides/http/url-generate#generating-signed-urls), which are non db backed URLs, but are not one time only.

In order to have one time only urls, you need to store the state on your server and cannot be done without it.

One way is to generate signed urls with small expiry, lets say 30mins and after they are used, you can store the signature in Redis and dis-allow it next time.

But, yes as far as AdonisJS is concerned, I have no plans of introducing this functionality in the core.
